odl/set/sets.py

```diff
@@ -428,10 +428,7 @@ def __hash__(self):
 
     def element(self, inp=None):
         """Return a real number from ``inp`` or from scratch."""
-        if inp is not None:
-            return float(inp)
-        else:
-            return 0.0
+        return float(getattr(inp, 'real', 0.0))
```

Comment:
Is this the way to go?

Reply:
I wouldn't do it this way. The issue is that this maps anything which hasn't got a `.real` attribute to `0.0`. Core types like `int` or `bool` will play nicely with this, but anything else that could be mapped to a float wouldn't. For instance strings or custom classes that implement `__float__`, but not `real`.

So I'd prefer this:
```python
if inp is None:
    return 0.0
else:
    return float(getattr(inp, 'real', inp))
```

Reply:
I hadn't thought about that... That is a nice solution!